### PR TITLE
update Ember docs to latest version

### PR DESCRIPTION
**Pull request by yads**
* fix ember parsing to handle current guides/docs

If you're updating an existing documentation to it's latest version, please ensure that you have:

- [X ] Updated the versions and releases in the scraper file
- [X ] Ensured the license is up-to-date and that the documentation's entry in the array in `about_tmpl.coffee` matches it's data in `self.attribution`
- [ X] Ensured the icons and the `SOURCE` file in <code>public/icons/*your_scraper_name*/</code> are up-to-date if the documentation has a custom icon
- [ X] Ensured `self.links` contains up-to-date urls if `self.links` is defined
- [X ] Tested the changes locally to ensure:
  - The scraper still works without errors
  - The scraped documentation still looks consistent with the rest of DevDocs
  - The categorization of entries is still good

### CODE REVIEW

File: lib/docs/scrapers/ember.rb [hunk 1]
```diff
@@ -20,13 +22,7 @@ class Ember < UrlScraper
...
     options[:trailing_slash] = false
 
-    options[:container] = ->(filter) do
-      if filter.base_url.path.start_with?('/api')
-        'main article'
-      else
-        'main'
-      end
-    end
+    options[:container] = 'main'
```
Comment (simon04) [inline lib/docs/scrapers/ember.rb:25]: Unwrap article for api (for reduced content, and to make various CSS selectors effective, such as `._page > h1:first-child`)

```suggestion
    options[:container] = ->(filter) do
      if filter.base_url.host.start_with?('api')
        'main article'
      else
        'main'
      endhost.start_with?('api')
    end
```

File: lib/docs/filters/ember/entries.rb [hunk 2]
```diff
@@ -2,58 +2,39 @@ module Docs
...
         css('section').each_with_object [] do |node, entries|
           next unless heading = node.at_css('h3[data-anchor]')
           next if node.at_css('.github-link').content.include?('Inherited')
-          name = heading.at_css('span').content.strip
 
-          # Give their own type to "Ember.platform", "Ember.run", etc.
-          if self.type != 'Data' && name.include?('.')
-            type = "#{self.name.remove(/ \(.*/)}.#{name.split('.').first}"
-          end
+          name = heading.at_css('span').content.strip
```
Comment (simon04) [inline lib/docs/filters/ember/entries.rb:37]: This is the list I get when searching for "checkbox". Only 3 out of x properties are shown. The entries Checkbox `(events)`, `.` and `.()` all link to /ember/classes/checkbox/methods.

Screenshot_2021-02-19 Ember js Checkbox — DevDocs

File: assets/javascripts/templates/pages/about_tmpl.coffee [hunk 3]
```diff
@@ -273,7 +273,7 @@ credits = [
     'https://www.gnu.org/licenses/gpl-3.0.html'
   ], [
     'Ember.js',
-    '2017 Yehuda Katz, Tom Dale and Ember.js contributors',
+    'Copyright (c) 2020 Yehuda Katz, Tom Dale and Ember.js contributors',
```
Comment (simon04) [inline assets/javascripts/templates/pages/about_tmpl.coffee:276]: We normally omit the "copyright (c)" for brevity.

```suggestion
    '2020 Yehuda Katz, Tom Dale and Ember.js contributors',
```

File: lib/docs/filters/ember/entries.rb (hunk 2 above)
Comment (simon04) [inline lib/docs/filters/ember/entries.rb:37]: Below "Classes", a gazillion non-working "Function" entries is shown:

Screenshot_2021-02-19 Ember js Classes — DevDocs

**Review by simon04 — CHANGES_REQUESTED** [review]
Hi, thank you for this pull request. I've added a few notes and suggestions. Most importantly, the "Classes" entries need some tuning.

File: lib/docs/scrapers/ember.rb (hunk 1 above)
Comment (yads) [reply lib/docs/scrapers/ember.rb:25]: The issue with unwrapping the content, is that not all pages in the Ember guides have an article. I originally wanted to have this be 'main article', but as it turned out not all pages have an article tag. There isn't much of a pattern, but I'll see if I can tune it a bit.

File: lib/docs/filters/ember/entries.rb (hunk 2 above)
Comment (yads) [reply lib/docs/filters/ember/entries.rb:37]: The reason that this is only listing items that do not have Inherited in one of the documentation fields (it's actually in the line above this one.

`          next if node.at_css('.github-link').content.include?('Inherited')`

I can get rid of that rule if that helps. I'll investigate the weird `.` and `.()` entries.

Comment (yads) [reply lib/docs/filters/ember/entries.rb:44]: Ok I fixed the `Checkbox.` issue, however, it seems like `Checkbox.()` is a bug with the Ember docs themselves. This entry is missing a name https://api.emberjs.com/ember/3.25/classes/Checkbox/methods?anchor= and is thereby coming out like that. I think I'd rather fix it in the Ember docs themselves than have a special case for this in the scraper.

Comment (yads) [reply lib/docs/filters/ember/entries.rb:44]: Fixed these and moved them into their own type

**Comment by yads** [comment]
@simon04 I think I've addressed most of the issues you've raised. I've also fixed the broken Packages as the links did not work properly.

File: lib/docs/scrapers/ember.rb (hunk 1 above)
Comment (yads) [reply lib/docs/scrapers/ember.rb]: I wound up reversing the original container rule as it works with the new docs better.

**Review by simon04 — APPROVED** [review]
Awesome, thank you!

**Pull request merged**